drivers/amlogic/media/vout/hdmitx/hdmi_tx_20/hdmi_tx_main.c

```diff
@@ -785,9 +785,9 @@ static int set_disp_mode_auto(void)
 					case HDMI_3840x2160p60_64x27:
 						if (para->cs == COLORSPACE_RGB444 || para->cs == COLORSPACE_YUV444)
 						{
-							para->cd = COLORDEPTH_24B;
-							pr_info("hdmitx: display colourdepth is forced to %d bits because of current video information code\n",
-								colour_depths[para->cd - COLORDEPTH_24B]);
+							para->cd = COLORDEPTH_36B;
+							para->cs = COLORSPACE_YUV422;
+							pr_info("hdmitx: display is forced to cs/cd YUV422 12-bit because of current video information code\n");
```

Comment:
That's not how the HDMI 2.0 spec works.
If the requested color space was RGB or 444, you can't just switch it to 422 12-bit.
Since not all TVs support this mode, and you can only output RGB/444 at 8-bit at this resolution and refresh rate, this change is invalid.

Reply:
Without the mode switch, any HDR10 content will result in YUV444 8-bit when playing @ 60hz.  This is indeed wrong.  You can see this happening with refresh rate switching set to off and the TV set to 60hz.

Reply:
Agree that it's wrong. We need to think of a solution that will fix the issue and will not cause regressions for other users.

Reply:
How about this instead @arthur-liberman :
```
--- a/drivers/amlogic/media/vout/hdmitx/hdmi_tx_20/hdmi_tx_main.c
+++ b/drivers/amlogic/media/vout/hdmitx/hdmi_tx_20/hdmi_tx_main.c
@@ -707,6 +707,24 @@
 				}
 				break;
 			default:
+				switch (vic & 0xff) {
+					case HDMI_3840x2160p50_16x9:
+					case HDMI_3840x2160p60_16x9:
+					case HDMI_4096x2160p50_256x135:
+					case HDMI_4096x2160p60_256x135:
+					case HDMI_3840x2160p50_64x27:
+					case HDMI_3840x2160p60_64x27:
+						if (para->cs == COLORSPACE_RGB444 || para->cs == COLORSPACE_YUV444)
+						{
+							if (hdev->rxcap.native_Mode & (1 << 4)) {
+								para->cs = COLORSPACE_YUV422;
+								pr_info("hdmitx: display is forced to YUV422 because of current video information code\n");
+							}
+						}
+						break;
+					default:
+						break;
+				}
 				break;
 		}
 		pr_info("hdmitx: display colour subsampling is auto set to %s (VIC: %d)\n",
@@ -775,25 +793,7 @@
 					else
 						para->cd = COLORDEPTH_24B;
 				}
-
-				switch (vic & 0xff) {
-					case HDMI_3840x2160p50_16x9:
-					case HDMI_3840x2160p60_16x9:
-					case HDMI_4096x2160p50_256x135:
-					case HDMI_4096x2160p60_256x135:
-					case HDMI_3840x2160p50_64x27:
-					case HDMI_3840x2160p60_64x27:
-						if (para->cs == COLORSPACE_RGB444 || para->cs == COLORSPACE_YUV444)
-						{
-							para->cd = COLORDEPTH_24B;
-							pr_info("hdmitx: display colourdepth is forced to %d bits because of current video information code\n",
-								colour_depths[para->cd - COLORDEPTH_24B]);
-						}
-						break;
-					default:
-						break;
 				break;
-			}
 		}
 		pr_info("hdmitx: display colourdepth is auto set to %d bits (VIC: %d)\n",
 			colour_depths[para->cd - COLORDEPTH_24B], vic);
```